### ISSUE TITLE
Hang after duration expires

### ISSUE THREAD

**Issue by gavv** [gavv/signal-estimator#5]
After duration expires, signal-estimator sometimes hangs (sleeps indefinitely) instead of exiting.

Good invocation example:

```
$ ./bin/signal-estimator -o pulse -i pulse -d 3
opening alsa writer for device pulse
suggested_latency: 8000 us
suggested_buffer_size: 384 samples
selected_buffer_time: 12000 us
selected_buffer_size: 576 samples
selected_period_time: 4000 us
selected_period_size: 192 samples
opening alsa reader for device pulse
suggested_latency: 8000 us
suggested_buffer_size: 384 samples
selected_buffer_time: 12000 us
selected_buffer_size: 576 samples
selected_period_time: 4000 us
selected_period_size: 192 samples
can't enable real-time scheduling policy
can't enable real-time scheduling policy
latency:  sw+hw  23.839ms  hw  15.845ms  hw_avg5  15.845ms
latency:  sw+hw  50.011ms  hw  35.244ms  hw_avg5  25.545ms
latency:  sw+hw  46.396ms  hw  39.304ms  hw_avg5  30.131ms
$
```

Bad invocation example:

```
$ ./bin/signal-estimator -o pulse -i pulse -d 2
opening alsa writer for device pulse
suggested_latency: 8000 us
suggested_buffer_size: 384 samples
selected_buffer_time: 12000 us
selected_buffer_size: 576 samples
selected_period_time: 4000 us
selected_period_size: 192 samples
opening alsa reader for device pulse
suggested_latency: 8000 us
suggested_buffer_size: 384 samples
selected_buffer_time: 12000 us
selected_buffer_size: 576 samples
selected_period_time: 4000 us
selected_period_size: 192 samples
can't enable real-time scheduling policy
can't enable real-time scheduling policy
latency:  sw+hw  19.113ms  hw  12.970ms  hw_avg5  12.970ms
latency:  sw+hw  43.722ms  hw  36.153ms  hw_avg5  24.562ms
<---------------- HANG
```

**Comment by gavv** [comment]
Fixed by df44a5bec0dad99d5ebf90f3a5b8a0e33bca0c22

**Comment by gavv** [comment]
Included into v0.0.6
